astropy/coordinates/tests/test_regression.py

```diff
@@ -751,3 +751,70 @@ def test_regression_10291():
     assert_quantity_allclose(
         venus.separation(sun), 554.427 * u.arcsecond, atol=0.001 * u.arcsecond
     )
+
+
+@pytest.mark.parametrize("coord_cls", [ICRS, SkyCoord])
+class TestRegression16998:
+    """Direct tests of the underlying problem causing gh-16998.
+
+    Note that the issue itself was of columns missing in data written to a file.
+    That is now tested directly by the "icrs" column in
+    astropy/io/tests/mixin_columns.py.
+    Here, we test the underlying problem, that .info._represent_as_dict()
+    did not return a full set of columns in some cases.
+    """
+
+    @classmethod
+    def setup_class(self):
+        self.ra = [0, 10] * u.deg
+        self.dec = [20, 30] * u.deg
+        self.pm_ra_cosdec = [40, 50] * u.mas / u.yr
+        self.pm_dec = [60, 70] * u.mas / u.yr
+        self.radial_velocity = [80, 90] * u.km / u.s
+
+    def get_kwargs(self, *args):
+        return {a: getattr(self, a) for a in args}
+
+    @pytest.mark.parametrize(
+        "extra_kwargs", [{}, dict(representation_type="unitspherical")]
+    )
+    def test_unitspherical_ra_dec(self, coord_cls, extra_kwargs):
+        kwargs = self.get_kwargs("ra", "dec")
+        coord = coord_cls(**kwargs, **extra_kwargs)
+        expected_entries = set(kwargs) | {"representation_type"}
```

Comment:
If you are already willing to let the tests generate `expected_entries` dynamically then why don't you replace the three test functions here with a single parametrized test? The big difference between this test and the other two is that in this case `expected_entries` does not contain `"differential_type"`, so it might be more convenient not to include this one in the parametrization, but combining the other two tests should be straightforward.

Reply:
You're right. I've now done it and it is a function again. Overall, perhaps a little less readable, but a lot shorter, so I think it is a win.